flink-staging/flink-streaming/flink-storm-compatibility/src/main/java/org/apache/flink/stormcompatibility/api/FlinkClient.java

```diff
@@ -0,0 +1,362 @@
+	
+	
+	/*
+	 * The following methods are derived from "backtype.storm.generated.Nimubs.Client"
+	 */
+	
+	/**
+	 * Parameter {@code uploadedJarLocation} is actually used to point to the local jar, because Flink does not support
+	 * uploading a jar file before hand. Jar files are always uploaded directly when a program is submitted.
+	 */
+	public void submitTopology(final String name, final String uploadedJarLocation, final String jsonConf, final FlinkTopology topology)
+		throws AlreadyAliveException, InvalidTopologyException {
+		this.submitTopologyWithOpts(name, uploadedJarLocation, jsonConf, topology, null);
+	}
+	
+	/**
+	 * Parameter {@code uploadedJarLocation} is actually used to point to the local jar, because Flink does not support
+	 * uploading a jar file before hand. Jar files are always uploaded directly when a program is submitted.
+	 */
+	@SuppressWarnings("unused")
+	public void submitTopologyWithOpts(final String name, final String uploadedJarLocation, final String jsonConf, final FlinkTopology topology, final SubmitOptions options)
+		throws AlreadyAliveException, InvalidTopologyException {
+		
+		// TODO improve by checking job manager, if running
+		if(this.topologyNameToJobId.containsKey(name)) {
+			throw new AlreadyAliveException();
+		}
+		
+		// TODO refactor and combine shared code (copied from RemoveStreamEnvironment)
+		try {
+			final JobGraph jobGraph = topology.getStreamGraph().getJobGraph(name);
+			
+			// TODO improve error handling for missing config values and uploadedJarLocation==null
+			final String stormCoreJar = (String)this.config.get(FLINK_STROM_CORE_JAR_LOCATION);
+			final String jsonSimpleJar = (String)this.config.get(FLINK_STROM_JSON_SIMPLE_JAR_LOCATION);
+			final String[] jarFileNames = new String[] {stormCoreJar, jsonSimpleJar, uploadedJarLocation};
+			
+			final List<File> jarFiles = new ArrayList<File>();
+			for(int i = 0; i < jarFileNames.length; i++) {
+				final File file = new File(jarFileNames[i]);
+				try {
+					JobWithJars.checkJarFile(file);
+				} catch(final IOException e) {
+					throw new RuntimeException("Problem with jar file " + file.getAbsolutePath(), e);
+				}
+				jarFiles.add(file);
+			}
+			
+			for(final File file : jarFiles) {
+				jobGraph.addJar(new Path(file.getAbsolutePath()));
+			}
+			
+			final Configuration configuration = jobGraph.getJobConfiguration();
+			final Client client = new Client(new InetSocketAddress(this.jobManagerHost, this.jobManagerPort),
+				configuration, JobWithJars.buildUserCodeClassLoader(jarFiles, JobWithJars.class.getClassLoader()), -1);
+			
+			try {
+				final JobSubmissionResult result = client.run(jobGraph, false);
+				this.topologyNameToJobId.put(name, result.getJobID());
+				this.topologies.put(name, topology);
+			} catch(final ProgramInvocationException e) {
+				throw new RuntimeException("Cannot execute job due to ProgramInvocationException", e);
+			}
+			
+		} catch(final Exception e) {
+			e.printStackTrace();
+		}
```

Comment:
This happens on the client side, but I would still log it instead of printing stack trace.